### PR TITLE
Refactor how attitudes and penalties are stored

### PR DESCRIPTION
**Pull request by TomGoodIdea**
## Refactor Details
Removed government IDs and replaced every structure that used them with `Government *` maps. The ID system was added by MZ himself in https://github.com/endless-sky/endless-sky/commit/49dc75cdb94972c45a49dd403277ea9fb292a424, ~~but I couldn't find any good way to combine it with https://github.com/endless-sky/endless-sky/pull/9438. If someone finds out how to solve that problem, I won't need this refactor so much.~~ The problem is solved, so it's only code readability vs performance.

## Testing Done
Checked the following things to make sure it didn't broke anything:
 - attitude toward (Pirate / other governments)
 - foreign penalties (Quarg / Efreti)
 - custom penalties (Remnant / Indigenous Lifeform)

and everything worked ~~hmm, suspicious~~

### CODE REVIEW

**Comment by Hurleveur** [comment]
Why doesnt it work well with https://github.com/endless-sky/endless-sky/pull/9438 ?
Also I suggested this https://github.com/endless-sky/endless-sky/pull/9438#discussion_r1353265965

File: source/Government.cpp [hunk 1]
```diff
@@ -465,10 +457,7 @@ double Government::AttitudeToward(const Government *other) const
 	if(other == this)
 		return 1.;
 
-	if(attitudeToward.size() <= other->id)
-		return 0.;
-
-	return attitudeToward[other->id];
+	return attitudeToward.count(other) ? attitudeToward.at(other) : 0.;
```
Comment (quyykk) [inline source/Government.cpp:460]: This does two lookups. Use `find` instead

**Review by quyykk — COMMENTED** [review]
@tehhowch any thoughts? Idk why MZ changed it to use the id system.

File: source/Government.cpp [hunk 2]
```diff
@@ -457,7 +457,7 @@ double Government::AttitudeToward(const Government *other) const
 	if(other == this)
 		return 1.;
 
-	return attitudeToward.count(other) ? attitudeToward.at(other) : 0.;
+	return attitudeToward.find(other) != attitudeToward.end() ? attitudeToward.at(other) : 0.;
```
Comment (Hurleveur) [inline source/Government.cpp:460]: Using the find you can only look for the attribute once by storing it in a variable

**Comment by quyykk** [comment]
We'll just close this for now, there's not much reason to change it especially if we don't what MZ did it like that (although I think we can probably change it).

**Pull request closed without merging**